warehouse/query-core/src/main/java/datawave/query/postprocessing/tf/FieldValue.java

```diff
@@ -0,0 +1,131 @@
+package datawave.query.postprocessing.tf;
+
+import datawave.util.StringUtils;
+import org.apache.accumulo.core.data.Key;
+import org.apache.hadoop.io.Text;
+
+/**
+ * A field name and value which is sorted on {@code <value>\0<name>}
+ */
+@Deprecated
```

Comment:
Do you mean for this to be deprecated?

Reply:
Short answer yes. Long answer, the FieldValue class was a private static class within the TermFrequencyIterator. This merge request moves it out of the way for a lighter weight parsing class, TFKey

Reply:
Ah I see, thanks for explaining. Feel free to disregard any of my comments about this class if it's being phased out.

Reply:
Actually, if you want to take a look at TFKey and drop any comments on there that would be great. Always happy to have a second set of eyes on new code

Reply:
The main thing that jumps out at me is `TFKey.parse()`. I feel it would be more intuitive for this to be a static method that returns a new TFKey, rather than modify an existing TFKey. But that's mainly just a design preference.